app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  skip_before_action :require_login, only: [:activate, :confirm]
+  layout "login", only: [:activate, :confirm]
+
+  def index
+    @user = User.new
+    @users = User.all
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.save
+  end
+
+  def activate
+    @user = User.load_from_activation_token(params[:token])
+    not_authenticated unless @user.present?
+  end
+
+  def confirm
+    if @user = User.load_from_activation_token(params[:token])
+      if @user.update_attributes(confirmation_params)
```

Comment:
Can we call this `@user.update(contirmation_params)`? It does exactly the same as `update_attributes`, but I like `update` better :dancers: